src/client.ts

```diff
@@ -41,72 +43,40 @@ export type AcmeGroup = Opaque<any, 'AcmeGroup'>;
  * resources.
  */
 export class APIClient {
-  constructor(readonly config: IntegrationConfig) {}
+  readonly hubspot: Hubspot;
+  constructor(readonly config: IntegrationConfig) {
+    this.hubspot = new Hubspot(config.apiBaseUrl, config.apiKey);
+  }
 
   public async verifyAuthentication(): Promise<void> {
-    // TODO make the most light-weight request possible to validate
-    // authentication works with the provided credentials, throw an err if
-    // authentication fails
-    const request = new Promise<void>((resolve, reject) => {
-      http.get(
-        {
-          hostname: 'localhost',
-          port: 443,
-          path: '/api/v1/some/endpoint?limit=1',
-          agent: false,
-          timeout: 10,
-        },
-        (res) => {
-          if (res.statusCode !== 200) {
-            reject(new Error('Provider authentication failed'));
-          } else {
-            resolve();
-          }
-        },
-      );
-    });
-
     try {
-      await request;
+      const owners = await this.hubspot.get('/crm/v3/owners');
+      if (!Array.isArray(owners)) {
+        throw new Error('Provider authentication failed');
+      }
     } catch (err) {
       throw new IntegrationProviderAuthenticationError({
         cause: err,
-        endpoint: 'https://localhost/api/v1/some/endpoint?limit=1',
+        endpoint: `/crm/v3/properties/contact`,
         status: err.status,
         statusText: err.statusText,
       });
     }
   }
 
-  /**
-   * Iterates each user resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
-   */
-  public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
-      await iteratee(user);
+  public async iterateOwners(iteratee: ResourceIteratee<Owner>) {
+    try {
+      const owners = await this.hubspot.get('/crm/v3/owners');
+      for (const owner of owners) {
+        await iteratee(owner);
+      }
```

Comment:
This needs to be changed a bit. The main idea behind iterating resources in these functions rather than returning arrays of them is so that we don't hold too many objects in the memory at any given time. Sometimes the customer can have millions of these resources (not particularly for type 'Owners' but for something else :slightly_smiling_face: ).

Take a look here:
https://github.com/JupiterOne/graph-rapid7/blob/1dc194669abdf5cce8eada59cad1f67c743f0efc/src/client.ts#L194
https://github.com/JupiterOne/graph-rapid7/blob/1dc194669abdf5cce8eada59cad1f67c743f0efc/src/client.ts#L71

Also, you should modify your ".get" function to be generic so that we can use it for everything and be able to pass the expected return type.